src/renderer/services/thorchain/thorNode.ts

```diff
@@ -103,6 +117,68 @@ const getNodeInfo$ = (node: Address, network: Network): NodeInfoLD =>
     RxOp.startWith(RD.pending)
   )
 
+const { stream$: reloadLiquidityProviders$, trigger: reloadLiquidityProviders } = triggerStream()
+
+const getLiquidityProviders = ({ asset, network, assetDecimal }: GetLiquidityProvidersParams): LiquidityProvidersLD => {
+  const poolString = assetToString(asset)
+  return FP.pipe(
+    reloadLiquidityProviders$,
+    RxOp.switchMap(() => thorNodeApiAddress$(network)),
+    liveData.chain((api) =>
+      FP.pipe(
+        api,
+        (thorApi) => RxAjax.ajax.getJSON(`${thorApi}/pool/${poolString}/liquidity_providers`),
+        // We can not use something like t.array(LiquidityProviderIO) as in case if one of elements
+        // fails validation the WHOLE array will be processed as failed-validation by io-ts.
+        // So we need to validate each elemnt "by-hands" to avoid losing all data 'cuz of 1 element's fail
+
+        // First check - check if we have an array itself at the response
+        RxOp.map(t.array(t.unknown).decode),
+        RxOp.map(E.fold(() => [], FP.identity)),
+        // Secondly - check every response's element with LiquidityProviderIO codec and filter out every failed-validation-element
+        RxOp.map(A.filterMap(FP.flow(LiquidityProviderIO.decode, O.fromEither))),
+        RxOp.map(RD.success),
+        liveData.map(
+          A.map((provider) => ({
+            asset: provider.asset,
+            address: provider.rune_address,
+            pendingRune: baseAmount(provider.pending_rune, THORCHAIN_DECIMAL),
+            pendingAsset: baseAmount(provider.pending_asset, assetDecimal),
+            runeDepositValue: baseAmount(provider.rune_deposit_value, THORCHAIN_DECIMAL),
+            assetDepositValue: baseAmount(provider.asset_deposit_value, assetDecimal)
```

Comment:
Do we need `runeDepositValue` + `assetDepositValue`?

Reply:
no. left this for possible future needs. can remove that. np

Reply:
Yep, let's remove to have things we need only.